### DISCUSSION TITLE
Using MONAI FROC metric for bounding box based lesion detection without GT lesion mask.

### DISCUSSION THREAD

**Discussion by shivasv97** [Project-MONAI/MONAI#6648] [Q&A]
Hello,

I am trying to use the MONAI FROC metric (https://docs.monai.io/en/stable/metrics.html#froc) for calculating the FROC score and curve for my model which outputs bounding box coordinates(xmin, ymin, xmax, ymax). 
My ground truth is also bounding box values(xmin, ymin, xmax, ymax). The model outputs bounding boxes for multiple classes, and the ground truth contains multiple bounding boxes for any given class.

In the MONAI API link above, would it support my predictions and ground truths? Since, it looks more focussed towards the CAMELYON challenge format, as the challenge doesn't seem to have bounding box based outputs. 

If anyone used this FROC API for bounding box based evaluation, please let me know how you did so. Since there is no explicit way to define IoU threshold values and bounding box confidence thresholds, etc.

**Comment by KumoLiu** [comment]
Hi @shivasv97, I think you could refer to `LesionFROC` by using `monai.utils.prob_nms.ProbNMS`.
https://github.com/Project-MONAI/MONAI/blob/70e2151b602287e9e15e36c22ea9d061e30a60aa/monai/apps/pathology/metrics/lesion_froc.py#L37

Thanks!

**Reply by shivasv97** [reply]
Thank you for your reply @KumoLiu ! 
I went through the source code of ProbNMS. However, I am having trouble in understanding how to use it for my purpose. Could you please explain to help me understand?

**Reply by KumoLiu** [reply]
Hi @shivasv97, let me clarify, maybe what I suggest can't solve your case. I want to say that if you have a probability map, you can use `ProbNMS` to generate a list of selected lists, where inner lists contain probability and coordinates. Then you can follow the LesionFROC steps, using these probs and coordinates to calculate the FROC.
Hope it can help you, thanks!
